contracts/gateway/OSTComposer.sol

```diff
@@ -0,0 +1,437 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// ----------------------------------------------------------------------------
+//
+// http://www.simpletoken.org/
+//
+// ----------------------------------------------------------------------------
+
+import "../utilitytoken/contracts/organization/contracts/Organized.sol";
+import "./StakerProxy.sol";
+import "./EIP20GatewayInterface.sol";
+import "../lib/EIP20Interface.sol";
+import "../lib/Mutex.sol";
+import "openzeppelin-solidity/contracts/math/SafeMath.sol";
+
+/**
+ * @title OSTComposer implements Organized contract. Reentrancy is prevented
+ *        by using Mutex contract.
+ *
+ * @notice It facilitates the staker to get the OSTPrime on sidechains.
+ */
+contract OSTComposer is Organized, Mutex {
+
+     /* Usings */
+
+    using SafeMath for uint256;
+
+
+    /* Constants */
+
+    bytes32 constant public STAKEREQUEST_INTENT_TYPEHASH = keccak256(
+        abi.encode(
+            "StakeRequest(address gateway,uint256 amount,address staker,address beneficiary,uint256 gasPrice,uint256 gasLimit, uint256 nonce)"
+        )
+    );
+
+
+    /* Events */
+
+    /** Emitted whenever a request stake is called. */
+    event StakeRequested(
+        uint256 amount,
+        address beneficiary,
+        uint256 gasPrice,
+        uint256 gasLimit,
+        uint256 nonce,
+        address indexed staker,
+        address gateway,
+        bytes32 stakeRequestHash
+    );
+
+    /** Emitted whenever a revoke stake is called. */
+    event StakeRevoked(
+        address indexed staker,
+        bytes32 stakeRequestHash
+    );
+
+    /** Emitted whenever a reject stake is called. */
+    event StakeRejected(
+        address indexed staker,
+        bytes32 stakeRequestHash
+    );
+
+
+    /* Struct */
+
+    /**
+     * StakeRequest stores the stake amount, beneficiary address, gas price, gas
+     * limit, nonce, staker and gateway address.
+     */
+    struct StakeRequest {
+
+        /** Amount that will be staked. */
+        uint256 amount;
+
+        /**
+         * Address where the utility tokens will be minted in the
+         * auxiliary chain.
+         */
+        address beneficiary;
+
+        /** Gas price that staker is willing to pay. */
+        uint256 gasPrice;
+
+        /** Gas limit that staker is willing to pay. */
+        uint256 gasLimit;
+
+        /** Staker nonce at the gateway. */
+        uint256 nonce;
+
+        /** Address of the staker. */
+        address staker;
+
+        /** Address of the gateway where amount will be staked. */
+        EIP20GatewayInterface gateway;
+    }
+
+
+    /* Public Variables */
+
+    /* Mapping of staker to gateway to store stake request hash. */
+    mapping (address => mapping(address => bytes32)) public stakeRequestHashes;
+
+    /* Mapping of staker addresses to their StakerProxy. */
+    mapping (address => StakerProxy) public stakerProxies;
```

Comment:
I think both mappings can be combined to a single object as both are details about a staker. 
```
mapping (address => StakerProxy) public stakerProxies;
mapping(address => uint256) public activeStakeRequestCount;
```
Also They are accessed together in `removeStakerProxy`.

Reply:
Same as above.